### PR TITLE
Update Harbor OVA/SSO config, generate passwords

### PR DESCRIPTION
**Pull request by andrewtchin**
- Remove password options from OVA deploy wizard
- Set random `db_password` and `clair_db_password` on first boot
Fixes https://github.com/vmware/vic/issues/4727

### CODE REVIEW

File: installer/packer/scripts/harbor/start_harbor.sh [hunk 1]
```diff
@@ -40,10 +40,4 @@ function gc {
   echo "===================================================="
 }
 
-gc_enabled=$(ovfenv --key registry.gc_enabled)
```
Comment (reasonerjt) [inline installer/packer/scripts/harbor/start_harbor.sh]: I think everything related to gc should remain the same.

File: installer/packer/scripts/harbor/configure_harbor.sh [hunk 2]
```diff
@@ -43,22 +43,50 @@ ca_cert="${cert_dir}/ca.crt"
 ca_key="${cert_dir}/ca.key"
 ext="${cert_dir}/extfile.cnf"
 
+MANAGED_KEY="# Managed by configure_harbor.sh"
+export LANGUAGE="en_US"
+export LC_ALL="C"
```
Comment (hmahmood) [inline installer/packer/scripts/harbor/configure_harbor.sh]: I think this is going to overwrite the `LANGUAGE` setting above.

File: installer/packer/scripts/harbor/configure_harbor.sh [hunk 3]
```diff
@@ -189,13 +216,9 @@ configureHarborCfg ssl_cert $cert
...
 
-for attr in "${attrs[@]}"
-do
-  echo "Read attribute using ovfenv: [ $attr ]"
-  value=$(ovfenv -k $attr)
-
-  configureHarborCfg $(echo ${attr} | cut -d. -f2) "$value"
-done
+# Set MySQL and Clair DB passwords on first boot
+configureHarborCfgOnce db_password $(genPass)
```
Comment (hmahmood) [inline installer/packer/scripts/harbor/configure_harbor.sh]: Do we have the user/group ownership/permissions on the harbor config file set appropriately? By that I mean it should only be group/user readable by the same group/user harbor runs under.

File: installer/packer/scripts/harbor/start_harbor.sh (hunk 1 above)
Comment (andrewtchin) [inline installer/packer/scripts/harbor/start_harbor.sh]: I removed the part where we try to set it in `harbor.cfg` and set the default value to true in the OVA deploy

File: installer/packer/scripts/harbor/configure_harbor.sh (hunk 2 above)
Comment (andrewtchin) [inline installer/packer/scripts/harbor/configure_harbor.sh]: Good call, fixed

File: installer/packer/scripts/harbor/configure_harbor.sh (hunk 3 above)
Comment (andrewtchin) [inline installer/packer/scripts/harbor/configure_harbor.sh]: Added check for config file perms

File: installer/packer/vic-unified.ovf [hunk 4]
```diff
@@ -237,24 +237,16 @@ EVALUATION LICENSE. If You are licensing the Software for evaluation purposes, Y
...
-        <Label>3.4. Registry Admin Password</Label>
-        <Description>The initial password of registry admin. Subsequent changes of the admin password should be performed in UI.  (8-20 characters)</Description>
-      </Property>
-      <Property ovf:key="db_password" ovf:password="true" ovf:qualifiers="MinLen(8),MaxLen(128)" ovf:type="string" ovf:userConfigurable="true" ovf:value="">
-        <Label>3.5. Database Password</Label>
-        <Description>The initial password of the root user of MySQL database. Subsequent changes of password should be performed in operating system. (8-128 characters)</Description>
-      </Property>
-      <Property ovf:key="gc_enabled" ovf:type="boolean" ovf:userConfigurable="true" ovf:value="false">
-        <Label>3.6. Garbage Collection</Label>
+      <Property ovf:key="gc_enabled" ovf:type="boolean" ovf:userConfigurable="true" ovf:value="true">
```
Comment (frapposelli) [inline installer/packer/vic-unified.ovf]: Keep this false by default

Comment (andrewtchin) [inline installer/packer/vic-unified.ovf]: fixed

**Pull request merged**